Java/core/src/main/java/com/amazon/randomcutforest/tree/AbstractRandomCutTree.java

```diff
@@ -216,6 +218,33 @@ abstract protected NodeReference addNode(NodeReference parent, NodeReference lef
 
     abstract protected void deleteSequenceIndex(NodeReference node, long uniqueSequenceNumber);
 
+    abstract boolean modifyBoxAndCheckContains(NodeReference tempNode, Point point);
+
+    abstract void readjustPointSum(NodeReference node, Point point);
+
+    void recursivelyReadjustPointSum(NodeReference node, Point point) {
+        if (enableCenterOfMass) {
+            NodeReference tempNode = node;
+            while (tempNode != null) {
+                readjustPointSum(tempNode, point);
+                tempNode = getParent(tempNode);
+            }
+        }
+    }
+
+    // manages the bounding boxes and center of mass
+    void updateAncestorNodesAfterDelete(NodeReference nodeReference, Point point) {
+        NodeReference tempNode = nodeReference;
+        boolean boxNeedsUpdate = enableCache;
+        while (boxNeedsUpdate && tempNode != null) {
+            boxNeedsUpdate = !modifyBoxAndCheckContains(tempNode, point);
```

Comment:
OK, I see why you used `modify` now, but I think we should either use `update` to be consistent with this method, or else `compute` or `recompute` to indicate that we're computing a deterministic function. And then putting a comment on that function to explain that it's called after something has changed in the subtree would be helpful.

Reply:
this corresponds to a recompute version of getReflatedBox ... so added a flag to getRelated and reused that, along with contains.